code/datums/components/blind_sense.dm

```diff
@@ -0,0 +1,87 @@
+/datum/component/blind_sense
+	///The range we can hear-ping things from
+	var/hear_range = 8
+	///List of things we can't sense
+	var/list/sense_blacklist
+	///The amount of time you can sense things for
+	var/sense_time = 5 SECONDS
+	///Reference to the users ears
+	var/obj/item/organ/ears/ears
+	///Type casted to access client, owner
+	var/mob/owner
+	///What texture we use
+	var/texture = "texture_2"
+	///Do we bloom?
+	var/bloom = TRUE
+
+/datum/component/blind_sense/New(list/raw_args)
+	. = ..()
+	//Register signal for sensing voices
+	RegisterSignal(SSdcs, COMSIG_GLOB_LIVING_SAY_SPECIAL, .proc/handle_hear)
+	//Register signal for sensing sounds
+	RegisterSignal(SSdcs, COMSIG_GLOB_SOUND_PLAYED, .proc/handle_hear)
+	//typecast to access client
+	owner = parent
+	//Register ears for people with them - deaf people can't use this component
+	if(iscarbon(owner))
+		var/mob/living/carbon/C = owner
+		ears = locate(/obj/item/organ/ears) in C.internal_organs
+		RegisterSignal(ears, COMSIG_PARENT_QDELETING, .proc/handle_ears)
+
+/datum/component/blind_sense/RemoveComponent()
+	. = ..()
+	qdel(src)
+
+/datum/component/blind_sense/proc/handle_hear(datum/source, atom/speaker, message)
+	SIGNAL_HANDLER
+
+	//Stop us from hearing ourselves or if we're deaf
+	if(owner == speaker || ears?.deaf)
+		return
+	//Preset types for things without icons / fucky icons
+	var/type
+	if(isliving(speaker))
+		type = "mob"
+		if(isfile(message)) //soundfile exclusively from footstep signal
```

Comment:
doesnt work on filepaths. only works on `file()` objects.

Reply:
It just works
![image](https://user-images.githubusercontent.com/40559528/208799610-204b6a98-0516-4479-9cfd-206b36a583c9.png)
(No really, it does)

Reply:
Isn't this right, or am I missing what you said?
![image](https://user-images.githubusercontent.com/40559528/208799823-9fc66590-fb0c-4c10-af60-5b70d16d2912.png)
![image](https://user-images.githubusercontent.com/40559528/208800698-8a869b2a-c906-4611-ac57-19c79cb18d14.png)

Reply:
Apparently it does. Coulda sworn I tried to do the same a few months ago. Oh well!